provers/stark/src/table.rs

```diff
@@ -0,0 +1,108 @@
+use lambdaworks_math::field::{element::FieldElement, traits::IsFFTField};
+
+/// A two-dimensional Table holding field elements, arranged in a row-major order.
+/// This is the basic underlying data structure used for any two-dimensional component in the
+/// the STARK protocol implementation, such as the `TraceTable` and the `EvaluationFrame`.
+/// Since this struct is a representation of a two-dimensional table, all rows should have the same
+/// length.
+#[derive(Clone, Default, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+pub struct Table<F: IsFFTField> {
+    pub data: Vec<FieldElement<F>>,
+    pub width: usize,
+    pub height: usize,
+}
+
+impl<F: IsFFTField> Table<F> {
+    /// Crates a new Table instance from a one-dimensional array in row major order
+    /// and the intended width of the table.
+    pub fn new(data: &[FieldElement<F>], width: usize) -> Self {
+        // Check if the intented width is 0, used for creating an empty table.
+        if width == 0 {
+            return Self {
+                data: Vec::new(),
+                width,
+                height: 0,
+            };
+        }
+
+        // Check that the one-dimensional data makes sense to be interpreted as a 2D one.
+        debug_assert!(crate::debug::check_2d_array_integrity(data, width));
+        let height = data.len() / width;
```

Comment:
Should this be a debug_assert ? Shouldn't this raise an error ?

Reply:
Also, it's not clear what this function does. Maybe the name could be better

Reply:
I agree with the debug assertion, for the moment I can use a normal `assert!()` until we have better error handling.

Reply:
I renamed the function to `validate_2d_structure`, what do you think?

Reply:
Ok, we. can think of a better name later